src/agent/src/rpc.rs

```diff
@@ -1906,6 +1917,91 @@ mod tests {
         assert!(result.is_err(), "expected add arp neighbors to fail");
     }
 
+    #[tokio::test]
+    async fn test_get_memory_info() {
+        #[derive(Debug)]
+        struct TestData<'a> {
+            // if None is provided, no files will be generated, else the data in the Option will populate the file
+            block_size_data: Option<&'a str>,
+            hotplug_probe_data: Option<&'a str>,
+
+            result: (u64, bool),
+            error_contains: &'a str,
+        }
+
+        let tests = &[
+            TestData {
+                block_size_data: Some("10000000"),
+                hotplug_probe_data: Some("hotplug data"),
+                result: (268435456, true),
+                error_contains: "",
+            },
+            TestData {
+                block_size_data: Some("100"),
+                hotplug_probe_data: None,
+                result: (256, false),
+                error_contains: "",
+            },
+            TestData {
+                block_size_data: None,
+                hotplug_probe_data: None,
+                result: (0, false),
+                error_contains: "",
+            },
+            TestData {
+                block_size_data: Some(""),
+                hotplug_probe_data: None,
+                result: (0, false),
+                error_contains: "Invalid block size",
+            },
+            TestData {
+                block_size_data: Some("-1"),
+                hotplug_probe_data: None,
+                result: (0, false),
+                error_contains: "Invalid block size",
+            },
+            TestData {
+                block_size_data: Some("    "),
+                hotplug_probe_data: None,
+                result: (0, false),
+                error_contains: "Invalid block size",
+            },
+        ];
+
+        for (i, d) in tests.iter().enumerate() {
+            let msg = format!("test[{}]: {:?}", i, d);
+
+            let dir = tempdir().expect("failed to make tempdir");
+            let block_size_path = dir.path().join("block_size_bytes");
+            let hotplug_probe_path = dir.path().join("probe");
+
+            if let Some(block_size_data) = d.block_size_data {
+                fs::write(&block_size_path, block_size_data).unwrap();
+            }
+            if let Some(hotplug_probe_data) = d.hotplug_probe_data {
+                fs::write(&hotplug_probe_path, hotplug_probe_data).unwrap();
```

Comment:
Nit: `hotplug_probe_path` is only checked for existence, so `hotplug_probe_data` can be a bool type, and if true then touch an empty file.